genesyscloud/resource_genesyscloud_telephony_didpool_test.go

```diff
@@ -0,0 +1,115 @@
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				// Create
+				Config: generateDidPoolResource(
+					didPoolResource1,
+					didPoolStartPhoneNumber1,
+					didPoolEndPhoneNumber1,
+					nullValue, // No description
+					nullValue, // No comments
+					nullValue, // No provider
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "start_phone_number", didPoolStartPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "end_phone_number", didPoolEndPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "description", ""),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "comments", ""),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "pool_provider", ""),
+				),
+			},
+			{
+				// Update
+				Config: generateDidPoolResource(
+					didPoolResource1,
+					didPoolStartPhoneNumber1,
+					didPoolEndPhoneNumber1,
+					strconv.Quote(didPoolDescription1),
+					strconv.Quote(didPoolComments1),
+					strconv.Quote(didPoolProvider1),
+				),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "start_phone_number", didPoolStartPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "end_phone_number", didPoolEndPhoneNumber1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "description", didPoolDescription1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "comments", didPoolComments1),
+					resource.TestCheckResourceAttr("genesyscloud_telephony_did_pool."+didPoolResource1, "pool_provider", didPoolProvider1),
+				),
+			},
+			{
+				// Import/Read
+				ResourceName:      "genesyscloud_telephony_did_pool." + didPoolResource1,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testVerifyDidPoolsDestroyed,
+	})
+}
+
+func generateDidPoolResource(
+	resourceID string,
+	startPhoneNumber string,
+	endPhoneNumber string,
+	description string,
+	comments string,
+	poolProvider string) string {
+	return fmt.Sprintf(`resource "genesyscloud_telephony_did_pool" "%s" {
+		start_phone_number = "%s"
+		end_phone_number = "%s"
+		description = %s
+		comments = %s
+		pool_provider = %s
+	}
+	`, resourceID, startPhoneNumber, endPhoneNumber, description, comments, poolProvider)
+}
+
+func testVerifyDidPoolsDestroyed(state *terraform.State) error {
+	telephonyAPI := platformclientv2.NewTelephonyProvidersEdgeApi()
+	for _, rs := range state.RootModule().Resources {
+		if rs.Type != "genesyscloud_telephony_did_pool" {
```

Comment:
From a readability perspective, GoLang is very big on avoiding nest if.. then..else statements.  Instead, look for how you can refactor the code below to use a single line of if..then statements that either break or return.  You should be able to see a straight line all the way down your code with your conditionals.  Take a look at this blog piece.

https://www.geeksforgeeks.org/writing-clean-else-statements/

Reply:
Thanks for the tip John, I agree it's more readable that way. I've refactored them to separate 'if block' statements